modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -659,6 +659,23 @@ public Map<String, ModelsMap> updateAllModels(Map<String, ModelsMap> objs) {
         return objs;
     }
 
+    /**
+     * Removes importToRemove from the imports of objs, if present.
+     * This is useful to remove imports that are already present in operations-related template files, to avoid importing the same thing twice.
+     *
+     * @param objs imports will be removed from this objs' imports collection
+     * @param importToRemove the import statement to be removed
+     */
+    protected void removeImport(OperationsMap objs, String importToRemove) {
```

Comment:
instead of just removing a "known" duplicate, what about simply looping through imports to remove all duplicates?

ideally all imports should be added via a function to check for duplicates first before inserting.

Reply:
> instead of just removing a "known" duplicate, what about simply looping through imports to remove all duplicates?

I'm not sure I get it. My understanding is this:
Duplicate imports arise if one of the two import statements is added dynamically from code, and the other of the two import statements is hard-coded in a mustache file.
We do already make sure that we don't add the same import twice _dynamically from code_. This is simply done by having imports in a `Set` (see e.g. [here](https://github.com/OpenAPITools/openapi-generator/blob/0e212f53f2702512ed9b406eb5046855308c59b6/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L5666)).
In `removeImport` - or any other part of the code - we don't know what imports are hard-coded in the mustache files. So we can't have a generic solution that filters out all of them. (Or at least that would be a big change, migrating all hard-coded imports from mustache to dynamically generated ones.)

> ideally all imports should be added via a function to check for duplicates first before inserting.

Sounds good. I can change my code so that instead of removing certain imports, it prevents them from being added in the first place. Just want to check first whether we are on the same page about the topic above.

Reply:
thanks for the explanation.

> remove imports that are already present in operations-related template files

when I first read it, I didn't know you're referring to hardcoded import in the templates. all make sense now.